src/main/java/core/basesyntax/model/FruitTransaction.java

```
@@ -0,0 +1,34 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
```

Constructor lacks validation for input parameters. Please validate and throw informative RuntimeException when inputs are invalid. This violates checklist items: "Validate non-null parameters to public methods; throw informative RuntimeException on nulls.", "Validate fruit name is non-null/non-empty", and "Validate quantity parses to a non-negative integer; quantity less than zero is invalid and must cause an error". Consider throwing `FruitShopException` or `IllegalArgumentException` with a message that includes the offending parameter and value. Also ensure this class enforces invariants early so other services don't need to defensively check model state.